### PR TITLE
finished milestone 4

### PR DESCRIPTION
**Pull request by tafodinho**

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const CATEGORY_FILTER = 'CATEGORY_FILTER';
+
+const filterReducer = (state = '', action) => {
+  switch (action.type) {
+    case CATEGORY_FILTER: {
+      if (action.category === 'ALL') {
```
Comment (SunnyAsar) [inline src/reducers/filter.js]: lets set the intial state to all and let the default return that state(All).

File: src/containers/BookList.js [hunk 2]
```diff
@@ -4,17 +4,37 @@ import React from 'react';
...
-import { removeBook } from '../actions';
+import { removeBook, filterCategory } from '../actions';
+import CategoryFilter from '../components/CategoryFilter';
 
 const BookList = (props) => {
-  const { books } = props;
+  const { books, filter } = props;
+  const filteredBooks = books.filter((book) => {
+    if (filter !== '') {
+      return book.category === filter;
```
Comment (SunnyAsar) [inline src/containers/BookList.js]: let's do the filter at the state level so we are don't bring data we don't  need hear.
- you can have a method that accepts the books we and the current filter and then use `mapStateToProps` to  read the result of what the method has all ready filtered.

File: src/containers/BookList.js [hunk 3]
```diff
@@ -36,15 +50,15 @@ const BookList = (props) => {
 };
 
 const mapStateToProps = (state) => ({
-  books: state.books,
+  books: filterBooks(state.books, state.filter),
 });
 const mapDispatchToProps = (dispatch) => ({
```
Comment (zenott) [inline src/containers/BookList.js]: [Optional] We could use the shorter version of mapStateToProps here:
```
const mapDispatchToProps = { removeBook, filterCategory };
```

File: src/containers/BookList.js [hunk 4]
```diff
@@ -4,17 +4,31 @@ import React from 'react';
...
 import PropTypes from 'prop-types';
 import Book from '../components/Book';
-import { removeBook } from '../actions';
+import { removeBook, filterCategory } from '../actions';
+import CategoryFilter from '../components/CategoryFilter';
+import filterBooks from '../helper';
 
 const BookList = (props) => {
   const { books } = props;
+  const categories = ['ALL', 'Action', 'Biography', 'History', 'Agriculture', 'Kids', 'Poetry', 'Sci-Fi'];
```
Comment (zenott) [inline src/containers/BookList.js]: [Optional] Would be better to put categories in a separate file because we use it in multiple files.

**Pull request merged**